contracts/SpokePool.sol

```diff
@@ -286,7 +321,54 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
         uint256 relayerRefundId,
         MerkleLib.DestinationDistribution memory distributionLeaf,
         bytes32[] memory inclusionProof
-    ) public {}
+    ) public override {
+        // Check integrity of leaf structure:
+        require(distributionLeaf.chainId == chainId(), "Invalid chainId");
+        require(distributionLeaf.refundAddresses.length == distributionLeaf.refundAmounts.length, "invalid leaf");
```

Comment:
I dont think any of these checks are needed. If either of these fail then the proof will fail. it's somewhat redundant, no?

Reply:
I think its better to be safe in case the bots miss something, we won't get into weird state

Reply:
merging this comment: https://github.com/across-protocol/across-smart-contracts-v2/pull/24#discussion_r801314310